UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterItem.cs

```diff
@@ -1,7 +1,22 @@
-﻿using UnityEngine;
+﻿using UnityEditor;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.Serialization;
 
 
-	[ExecuteInEditMode]
+[ExecuteInEditMode]
 	public class RegisterItem : RegisterTile
 	{
+		public delegate void Crossed();
+		public event Crossed OnCrossed;
+
+		[HideInInspector]
+		public RegisterPlayer CrossedRegisterPlayer;
+
+		public void Cross(ref RegisterPlayer registerPlayer)
+		{
+			CrossedRegisterPlayer = registerPlayer;
```

Comment:
IMO storing it in a field is unnecessary, also do use UnityEvents instead of delegates and c# events, it's what we use throughout the codebase.
Just define `public class RegisterPlayerEvent : UnityEvent<RegisterPlayer>{}` somewhere and use it as event field type instead. Invocation would be like `OnCrossed?.Invoke(registerPlayer)` and that's it. 
Also not sure why you use `ref` here as well

Reply:
Changed it to a unityevent and removed ref.

Reply:
I mean, you don't need a `RegisterPlayer CrossedRegisterPlayer` field at all in this class. just use parametrized custom UnityEvent<RegisterPlayer> that would pass player to the invoked method, Slip() in your case